code/modules/ruins/lavalandruin_code/tomb.dm

```diff
@@ -0,0 +1,363 @@
+		if("bdance") //attacks diagonals
+			var/static/list/bdance_angles = list(45, -45, 135, -135)
+		if("surge") //fast attacks in the front, medium on sides and slow in the back.
+			for(var/i in front_angles)
+				var/turf/Tsurge1 = get_step(src_turf, turn(dir_to_target, i))
+				new /obj/effect/temp_visual/attackwarn/fast(Tsurge1, src)
+			for(var/i in side_angles)
+				var/turf/Tsurge2 = get_step(src_turf, turn(dir_to_target, i))
+				new /obj/effect/temp_visual/attackwarn(Tsurge2, src)
+			for(var/i in back_angles)
+				var/turf/Tsurge3 = get_step(src_turf, turn(dir_to_target, i))
+				new /obj/effect/temp_visual/attackwarn/slow(Tsurge3, src)
+
+/mob/living/simple_animal/hostile/cryptguard/leader/adjustHealth(amount, updating_health = TRUE, forced = FALSE)
+	if(woke == FALSE)
+		var/madeit = FALSE
+		for(var/mob/living/simple_animal/hostile/cryptguard/goon in guards)
+			if(goon.woke == TRUE && goon.stat != DEAD)
+				break
+			if(goon.stat != DEAD)
+				break
+			madeit = TRUE
+		if(madeit == TRUE)//if you're looking  to put this in a shitcode thread you might want to look a little closer ;)
+			say("Your hubris! It will be your downfall, silly mortal! I am the true defender of the necropolis!")
+			awaken()
+		return
+	..()
+
+/mob/living/simple_animal/hostile/cryptguard/leader/attack_hand(mob/user)
+	if(woke)
+		return
+	if(user.a_intent == INTENT_HARM)
+		return
+	var/isleep = TRUE
+	var/finalboss = TRUE
+	for(var/mob/living/simple_animal/hostile/cryptguard/goon in guards)
+		if(goon.woke == TRUE && goon.stat != DEAD)//if any of them are woke but not dead, then they don't need to be awakened. just return.
+			isleep = FALSE
+		if(goon.stat != DEAD)//if any of them are alive, then awaken them instead.
+			finalboss = FALSE
+	if(isleep == FALSE)
+		return
+	if(finalboss == FALSE)
+		say("You are trespassing.")
+		icon_state = "antikytherasword"
+		if(trapgate.open == TRUE)
+			trapgate.toggle_the_gate()
+		trapgate.locked = TRUE
+		sleep(5)
+		for(var/mob/living/simple_animal/hostile/cryptguard/goonstve in guards)
+			goonstve.say("...")
+			goonstve.awaken()
+	else
+		if(quest == FALSE)
+			say("A worthy combatant! If you're not intending on trying to kill me, return with proof you've gotten rid of the foul serpents that plague these lands and i'll reward you.")
+			quest = TRUE
+			return
+
+//LOOT (THE MECH)//
+
+/obj/mecha/combat/necropolis
+	desc = "The guardians of the necropolis, before the dragons arrived."
+	name = "\improper Antikythera"
+	icon_state = "antikythera"
+	step_in = 4
+	dir_in = 1 //Facing North.
+	max_integrity = 400
+	deflect_chance = 20
+	armor = list("melee" = 70, "bullet" = 10, "laser" = 10, "energy" = 10, "bomb" = 20, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 100) //durand stats, but much higher melee and lower bullet and laser
+	max_temperature = 30000
+	infra_luminosity = 8
+	force = 40
+	wreckage = /obj/structure/mecha_wreckage/necropolis
+
+/obj/mecha/combat/stone/loaded/Initialize()
```

Comment:
Necropolis, not stone. Additionally needs a loaded version defined somewhere first.